packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -120,6 +122,15 @@ async function isLambdaFunctionCodeOnlyChange(
               foundCodeDifference = true;
               s3Key = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
               break;
+            case 'ZipFile':
+              foundCodeDifference = true;
+              // We must create a zip package containing a file with the inline code
+              const rawZipFile = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
+              // file extension must be chosen depending on the runtime
+              const runtime: string = await evaluateCfnTemplate.evaluateCfnExpression(change.newValue.Properties!.Runtime);
```

Comment:
Is the `!` really required? Can we go without it?

```suggestion
              const runtime: string = await evaluateCfnTemplate.evaluateCfnExpression(change.newValue.Properties.Runtime);
```

Reply:
Also, let's call this variable `functionRuntime`, just to make it's very clear what runtime we are talking about here.

Reply:
`Properties` is possibly undefined so we need to care it.

https://github.com/aws/aws-cdk/blob/c7c51debabeb3ec1de4f4c2dc986dfdfac27756b/packages/%40aws-cdk/cloudformation-diff/lib/diff/types.ts#L475-L480

I replaced `!` to `?` to care when runtime is undefined, c.f. `s3-bucket-deployments.ts`  (idk if there's such a case though)

https://github.com/aws/aws-cdk/blob/c7c51debabeb3ec1de4f4c2dc986dfdfac27756b/packages/aws-cdk/lib/api/hotswap/s3-bucket-deployments.ts#L25-L28